### DISCUSSION TITLE
Why use `/app/app` instead of `/code/app` in the docker image?

### DISCUSSION THREAD

**Discussion by hanhantw** [tiangolo/uvicorn-gunicorn-fastapi-docker#312] [Questions] [answered]
Hi, 

I'm new here using FastAPI.

While trying to deploy the project on docker, I notice that `/code/app` is used  for  pure python image like `python:3.9` but `/app/app` for FastAPI image `tiangolo/uvicorn-gunicorn-fastapi:python3.9` in [the tutorial of how to deploy on docker](https://fastapi.tiangolo.com/deployment/docker/?h=docker). 

Is there any reason for this?  I feel that `/app/app` is more confusing than `/code/app`. We have two directories in different levels both named `app` and an application in `main.py` called `app` as well.  For some beginners, I feel it might be hard to distinguish which directory  `app` refers to or if it refers to the application while reading the commands or codes. Or am I the only one to feel so?

**Comment by billcrook** [comment]
> Or am I the only one to feel so?

No, it's awkward.

**Comment by tiangolo** [comment, marked as answer]
I agree! `/app/app` is more confusing, that's why in the new docs I suggest `/code/app`. :nerd_face: 

It was like that only for backwards compatibility. :sweat_smile: 

At some point, a long time ago, I thought it was a good idea, but then I realized it wasn't. :grimacing: 

Anyway, now that Uvicorn supports managing workers with `--workers`, including restarting dead ones, there's no need for Gunicorn. That also means that it's much simpler to build a Docker image from scratch now, I updated the docs to explain it.

Because of that, I deprecated this Docker image: https://github.com/tiangolo/uvicorn-gunicorn-fastapi-docker#-warning-you-probably-dont-need-this-docker-image
